Tasks/GoTool/gotool.ts

```diff
@@ -0,0 +1,131 @@
+///<reference path="../../definitions/node.d.ts" />
+
+import * as toolLib from 'vsts-task-tool-lib/tool';
+import * as tl from 'vsts-task-lib/task';
+import * as os from 'os';
+import * as path from 'path';
+
+let osPlat: string = os.platform();
+let osArch: string = os.arch();
+
+try {
+    tl.setResourcePath(path.join(__dirname, "task.json"));
+} catch (error) {
+    tl.setResult(tl.TaskResult.Failed, error);
+}
+
+async function run() {
+    try {
+        let versionSpec = tl.getInput('version', true);
```

Comment:
Since version is a string, better to add validations over input. Trimming input will also be helpful!

Reply:
We need not to do validation, as any wrong input by users anyways lead to task failure. But yes trimming should be file

Reply:
Done
changes incorporated in https://github.com/Microsoft/vsts-tasks/pull/6307